### ISSUE TITLE
Attenuation coefficient values still differ from the XCOM website

### ISSUE THREAD

**Issue by danielbon** [lynntf/nist-calculators#3]
the attenuation coefficient values still differ from the XCOM website by a factor ~6 to 8 (it varies). These are the values provided by the python script:
H2O -> [1, 8] [0.11189834407236524, 0.8881016559276347]
data :
[(1.0 keV, 2.06819634e-01, 6.58465443e-03, 5.14184594e+02, 0.00000000e+00, 0.00000000e+00, 5.14191179e+02, 5.14397999e+02)

These are the values on XCOM website:
1.0 keV | 1.372E+00 | 1.319E-02 | 4.076E+03 | 0.000E+00 | 0.000E+00 | 4.077E+03 | 4.076E+03

The attenuation coefficients are supposed to be given in cm2/g.

**Comment by lynntf** [comment]
This may be related to the way mixtures are computed. Single element data appears to properly match the XCOM website. Perhaps related to [this pull request](https://github.com/lynntf/nist-calculators/pull/1). Currently investigating.

**Comment by lynntf** [comment]
> The interaction coefficients and total attenuation coefficients for compounds are obtained as weighted sums over the corresponding coefficients for elements. XCOM automatically calculates the weight factors, i.e., the fractions by weights of the atomic constituents, from the chemical formula for the compound entered by the user. For mixtures, the user must enter the fractions by weight of the components.

From [https://physics.nist.gov/PhysRefData/Xcom/Text/chap3.html](https://physics.nist.gov/PhysRefData/Xcom/Text/chap3.html)

Not sure that the package is doing this...

**Comment by lynntf** [comment]
Problematic lines in `xcom.py`. Mixture is only accessing the first material `material.weights[0]` in its calculation.

```python
if len(material) == 1:
        # Single constituent material
        element = material.elements_by_Z[0]
        data = calculate_cross_section(element, energy)
        # Attenutaion coefficient = macro_cross_secction/denisty = \
        # = micro_cross_section/atom_weight[gr]
        # atom_weight[gr] = atom_weight[amu] / AVOGADRO
        atom_weight = MaterialFactory.get_element_mass(element)
        for name in data.dtype.names:
            if name != "energy": data[name] *= (_AVOGADRO / atom_weight)
        return data
    elif len(material) > 1:
        # Mixture of materials
        atom_weights_amu = MaterialFactory.get_elements_mass_list(material.elements_by_Z)
        data = calculate_cross_section(material.elements_by_Z[0], energy)
        for name in data.dtype.names:
            if name != "energy": data[name] *= ( material.weights[0] * _AVOGADRO / atom_weights_amu[0])

        for atom_weight_amu, element, weight in zip(atom_weights_amu[1:], material.elements_by_Z[1:], material.weights):
            temp = calculate_cross_section(element, energy)
            for name in data.dtype.names:
                if name != "energy": data[name] +=  temp[name] * weight * _AVOGADRO / atom_weight_amu
        return data
    else:
        raise Exception("Empty material")
```

Need to change to weighted average

**Comment by lynntf** [comment]
I believe that the issue has been fixed now with [commit 1dc011662869d6e2bd1dbbdf442482cb107c8b53](https://github.com/lynntf/nist-calculators/commit/1dc011662869d6e2bd1dbbdf442482cb107c8b53). The weighted average for the mixture had an off by one index.

```python
>>> xcom.calculate_attenuation(xcom.MaterialFactory.from_formula("H2O"), [1e3])
array([(1000., 1.3722503, 0.01319334, 4075.62851585, 0., 0., 4075.64170918, 4077.01395948)],
      dtype=[('energy', '<f8'), ('coherent', '<f8'), ('incoherent', '<f8'), ('photoelectric', '<f8'), ('pair_atom', '<f8'), ('pair_electron', '<f8'), ('total_without_coherent', '<f8'), ('total', '<f8')])
```
